fix a segfault in pdftex
reported by David Carlisle on tex-live
http://tug.org/pipermail/tex-live/2016-June/038664.html
proposed fix by Akira Kakuto
http://tug.org/pipermail/tex-live/2016-June/038666.html
originally on tex.sx
http://chat.stackexchange.com/transcript/message/30145736
---
 texk/web2c/pdftexdir/utils.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- texlive-bin.orig/texk/web2c/pdftexdir/utils.c
+++ texlive-bin/texk/web2c/pdftexdir/utils.c
@@ -870,7 +870,8 @@
     int size, len = 0;          /* to avoid warning about uninitialized use of len */
 
     boolean found = i < sub_match_count
-        && match_string != NULL && pmatch[i].rm_so >= 0 && i >= 0;
+        && match_string != NULL && pmatch[i].rm_so >= 0 && i >= 0
+        && pmatch[i].rm_eo >= pmatch[i].rm_so;
 
     if (found) {
         len = pmatch[i].rm_eo - pmatch[i].rm_so;
